src/components/TokenInput.tsx

```diff
@@ -144,6 +146,7 @@ function TokenInput({
             variant="body2"
             color={disabled ? "text.secondary" : "text.primary"}
             textAlign="end"
+            sx={{ display: showUSDprice ? "block" : "none" }}
```

Comment:
can you wrap this component with the `showUSDprice` boolean instead of hiding it using CSS?

```
{showUSDprice && (<Component/>)}
```